ca/certificate-authority.go

```diff
@@ -277,18 +281,32 @@ func (ca *CertificateAuthorityImpl) IssueCertificate(csr x509.CertificateRequest
 	commonName := ""
 	hostNames := make([]string, len(csr.DNSNames))
 	copy(hostNames, csr.DNSNames)
+
 	if len(csr.Subject.CommonName) > 0 {
 		commonName = strings.ToLower(csr.Subject.CommonName)
 		hostNames = append(hostNames, commonName)
-	} else if len(hostNames) > 0 {
-		commonName = strings.ToLower(hostNames[0])
-	} else {
+	}
+
+	if len(hostNames) == 0 {
 		err = core.MalformedRequestError("Cannot issue a certificate without a hostname.")
 		// AUDIT[ Certificate Requests ] 11917fa4-10ef-4e0d-9105-bacbe7836a3c
 		ca.log.AuditErr(err)
 		return emptyCert, err
 	}
 
+	if ca.forceCNFromSAN && commonName == "" {
+		commonName = hostNames[0]
```

Comment:
We lost a `strings.ToLower(hostNames[0])` here. I think it's still relevant because commonName gets passed straight into the SignRequest.